staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -189,7 +189,7 @@ type DomainSpec struct {
 	Memory *Memory `json:"memory,omitempty"`
 	// Machine type.
 	// +optional
-	Machine Machine `json:"machine,omitempty"`
+	Machine *Machine `json:"machine,omitempty"`
```

Comment:
What is the reason for this change?

Reply:
The reason are changes in the validation. I asked @iholder-redhat to make it a pointer (since it is optional anyways), because an empty `type` on the machine was not accepted. But I checked now our example VMIs:

```
---
apiVersion: kubevirt.io/v1
kind: VirtualMachineInstance
metadata:
  labels:
    special: vmi-nocloud
  name: vmi-nocloud
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: containerdisk
      - disk:
          bus: virtio
        name: cloudinitdisk
      - disk:
          bus: virtio
        name: emptydisk
    machine:
      type: ""
    resources:
      requests:
        memory: 128Mi
  terminationGracePeriodSeconds: 0
  volumes:
  - containerDisk:
      image: registry:5000/kubevirt/cirros-container-disk-demo:devel
    name: containerdisk
  - cloudInitNoCloud:
      userData: |
        #!/bin/sh

        echo 'printed from cloud-init userdata'
    name: cloudinitdisk
  - emptyDisk:
      capacity: 2Gi
    name: emptydisk
```

We have 

```
    machine:
      type: ""
```

in there and we accepted it. That means that we need  continue accepting it. @iholder-redhat I think that means reverting this part and adding the optional the type in Machine.

Reply:
Thanks for the explanation and yes we need to add it there.

Reply:
Hey @rmohr,

We did have
```
    machine:
      type: ""
````
But it actually was an unintended mistake as `type` is supposed to be mandatory according to our schema definitions. Seems like because of Go's bug (not enforcing mandatory fields properly) we didn't notice that our example VMIs are actually invalid according to our own definitions.

I guess we'll just adapt this mistake to not break backwards comparability? Can this mistake, i.e. machine without a defined type, cause problems in the future?

Reply:
> But it actually was an unintended mistake as `type` is supposed to be mandatory according to our schema definitions. Seems like because of Go's bug (not enforcing mandatory fields properly) we didn't notice that our example VMIs are actually invalid according to our own definitions.

I don't see "machine"  required in any spec on master. Where do you see that?

Reply:
> I guess we'll just adapt this mistake to not break backward comparability? Can this mistake, i.e. machine without a defined type, cause problems in the future?

You can make it optional. It is not beautiful, but that is fine since we anyway default it always. After the VMI went through the webhooks, there is no case where no machine type is set.